reports/tests.py

```diff
@@ -0,0 +1,508 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""Tests module for the invoices reports"""
+
+# Imports #####################################################################
+from collections import defaultdict
+from datetime import timedelta, datetime
+from unittest import mock
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.test import TestCase, override_settings
+from django.utils import timezone
+
+import ddt
+
+from instance.models.appserver import Status as AppServerStatus
+from instance.models.openedx_instance import OpenEdXInstance
+from instance.tests.base import WithUserTestCase
+from instance.tests.models.factories.openedx_appserver import make_test_appserver
+from instance.tests.models.factories.openedx_instance import OpenEdXInstanceFactory
+from pr_watch.models import WatchedPullRequest
+from pr_watch.tests.factories import WatchedForkFactory
+from reports.helpers import (
+    _normalize_date,
+    generate_charge_details,
+    generate_charges,
+    generate_watched_forks_instances,
+    get_billing_period,
+    get_instance_charges,
+)
+from userprofile.factories import OrganizationFactory
+
+# Tests #####################################################################
+
+
+@ddt.ddt
+class ReportViewTestCase(WithUserTestCase):
+    """
+    Tests for reports views functionality
+    """
+    url = 'reports:report'
+
+    def setUp(self):
+        super(ReportViewTestCase, self).setUp()
+
+        self.username = 'user3'
+        self.password = 'pass'
+        self.organization = OrganizationFactory.create(
+            name='Organization Name',
+            github_handle='organization_github_handle'
+        )
+
+    @ddt.data(
+        'user0',  # Anonymous User
```

Comment:
user0 isn't a valid user (see the users at base.py), I don't know how is this even working, maybe it redirects to some login form when it detects that the user doesn't exist.
You can use the same format used by the tests I linked to earlier, where they use `None` for anonymous and then do „if user: then login()“ (and if user is None, then there's no login and thus anonymous).